lib/buffer.js

```diff
@@ -692,50 +692,51 @@ Buffer.prototype[customInspectSymbol] = function inspect() {
 Buffer.prototype.inspect = Buffer.prototype[customInspectSymbol];
 
 Buffer.prototype.compare = function compare(target,
-                                            start,
-                                            end,
-                                            thisStart,
-                                            thisEnd) {
+                                            targetStart,
+                                            targetEnd,
+                                            sourceStart,
+                                            sourceEnd) {
   if (!isUint8Array(target)) {
     throw new ERR_INVALID_ARG_TYPE('target', ['Buffer', 'Uint8Array'], target);
   }
   if (arguments.length === 1)
     return _compare(this, target);
 
-  if (start === undefined)
-    start = 0;
-  else if (start < 0)
-    throw new ERR_INDEX_OUT_OF_RANGE();
+  if (targetStart === undefined)
+    targetStart = 0;
+  else if (targetStart < 0)
+    throw new ERR_OUT_OF_RANGE('targetStart', '>= 0', targetStart);
```

Comment:
We might also want add the maximum at the same time:

```js
throw new ERR_OUT_OF_RANGE('targetEnd', `>= 0 and <= ${target.length}`, targetEnd)
```

Reply:
The line here checks `targetStart` and not `targetEnd`. There values for `targetStart` (and `sourceStart`) can be greater than `target.length` (and `buf.length`).

```console
$ node
> var buf = Buffer.alloc(10)
undefined
> var target = Buffer.alloc(10)
undefined
> buf.compare(target, 1, 9, 100, 9)
-1
> buf.compare(target, 100, 9, 1, 9)
1
> 
```